Lib/dis.py

```diff
@@ -474,15 +479,24 @@ def print_instruction_line(self, instr, mark_as_current):
             print(file=self.file)
 
         fields = []
-        # Column: Source code line number
+        # Column: Source code locations information
         if lineno_width:
-            if instr.starts_line:
-                lineno_fmt = "%%%dd" if instr.line_number is not None else "%%%ds"
-                lineno_fmt = lineno_fmt % lineno_width
-                lineno = _NO_LINENO if instr.line_number is None else instr.line_number
-                fields.append(lineno_fmt % lineno)
+            if self.show_positions:
+                # reporting positions instead of just line numbers
+                if instr_positions := instr.positions:
+                    ps = tuple('?' if p is None else p for p in instr_positions)
+                    positions_str = f"{ps[0]}:{ps[2]}-{ps[1]}:{ps[3]}"
```

Comment:
With line numbers, we put a `_NO_LINENO` symbol (which is `--`) if line number is None. That might be nicer here too, rather than ?:?-?:?.

Do we have a test for the case where some instructions have locations and some not?

Reply:
Yes... it's `test_dis_with_some_positions`. Now that I've understood how `co_linetable` works, I can make it more precise to have better mixed-case btw. There is `test_dis_with_no_positions` which tests when there is no positions for *all* instructions.

I can add a case to `test_dis_with_some_positions` where an instruction has no position at all as well, it would test the `?:?-?:?' which would be transformed into `--`. 

By the way, should we have `--` or `-:-:-:-`? (I would prefer the first form I think)

Reply:
Yes, I agree '--' looks nicer than '-:-:-:-'`